core/src/main/java/jeeves/transaction/TransactionManager.java

```diff
@@ -72,6 +74,15 @@
 
             result = action.doInTransaction(transaction);
 
+            // GlobalExceptionManager stores the exception in a request attribute, to be processed to rollback the transaction
+            // as otherwise GlobalExceptionManager processes the exception and the code doesn't enter in the catch block
+            // to rollback the transaction
+            Throwable ex = (Throwable) RequestContextHolder.currentRequestAttributes().getAttribute("exception", RequestAttributes.SCOPE_REQUEST);
+
+            if (ex != null) {
+                doRollback(context, transactionManager, transaction);
```

Comment:
Need to indicate `rolledBack = true` before calling `doRollback`, as we are tracking what has been done to correctly log near the end of the method.

Reply:
Okay two options:

1) move this code to finally block; or
2) add a check in catch statement to avoid a second `doRollback` call (if an exception occurs during this one)